### PR TITLE
Replace vuex with pinia

### PR DESCRIPTION
**Pull request by shrinathprabhu**
Resolves [AR-1939](https://team-1624093970686.atlassian.net/browse/AR-1939).

## Changes

- Install pinia.
- Setup pinia and create stores in pinia in `/src/stores`.
- Replace vuex related methods and getters with pinia state and actions in components.
- Remove vuex stores in `src/store`
- Remove vuex setup and uninstall vuex
- Add [types](https://www.npmjs.com/package/@types/bytes) for [bytes](https://www.npmjs.com/package/bytes)
- Remove ComputedRef and Ref types wherever not required as types in Pinia are inferred

## Checklist

- [x] The branch name follows the format: `developer/AR-XXX-issue-name`.
- [x] The changes have been tested locally.

### CODE REVIEW

File: src/stores/index.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+import { createPinia } from 'pinia'
```
Comment (shrinathprabhu) [inline src/stores/index.ts:1]:
```js
import { createPinia as createStore } from 'pinia'
const store = createStore()
export default store
```

File: src/stores/app.store.ts [hunk 2]
```diff
@@ -0,0 +1,394 @@
...
+
+    const [value, unit] = calculatedUserLimit.split(' ') as [
+      number,
+      UserLimitUnit
+    ]
+    return { value, unit, isUnlimited: false }
+  } else {
+    return { isUnlimited: true }
+  }
+}
```
Comment (shrinathprabhu) [inline src/stores/app.store.ts:138]: Move this to utils

File: src/stores/app.store.ts [hunk 3]
```diff
@@ -0,0 +1,394 @@
...
+        return {
+          verifier: authType.verifier,
+          clientId: authType.clientId,
+          clientSecret: authType.clientSecret,
+          redirectURL: authType.redirectUri,
+        }
+      })
+
+      if (
+        state.auth.passwordless.javascriptOrigin &&
```
Comment (shrinathprabhu) [inline src/stores/app.store.ts:221]: Add this getter as a function to gateway service file

File: src/stores/auth.store.ts [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+export const useAuthStore = defineStore('auth', {
+  state: (): AuthState => ({
+    email: '',
+    accessToken: '',
+    name: '',
+    walletAddress: '',
+  }),
+  actions: {
+    updateAccessToken(access_token: string) {
```
Comment (shrinathprabhu) [inline src/stores/auth.store.ts:18]: Change access_token as accesToken

File: src/stores/toast.store.ts [hunk 5]
```diff
@@ -0,0 +1,48 @@
+import { defineStore } from 'pinia'
+
+const TOAST_TIMEOUT = 5600
```
Comment (shrinathprabhu) [inline src/stores/toast.store.ts:3]: Prefer 3000ms

File: src/stores/url.store.ts [hunk 6]
```diff
@@ -0,0 +1,19 @@
+import { defineStore } from 'pinia'
```
Comment (shrinathprabhu) [inline src/stores/url.store.ts:1]: Rename it to config.store instead of url.store

Comment (shrinathprabhu) [inline src/stores/url.store.ts:1]: Remove this store and maintain forwarder and rpcUrl in gateway service and export it from there

**Review by ajithranka — APPROVED** [review]
No other comments apart from the ones noted during the call.

File: src/stores/index.ts (hunk 1 above)
Comment (shrinathprabhu) [reply src/stores/index.ts]: Resolved at https://github.com/arcana-network/developer-dashboard/pull/119/commits/646523feaa87c8025de16dd54501308a006251ab

File: src/stores/app.store.ts (hunk 2 above)
Comment (shrinathprabhu) [reply src/stores/app.store.ts]: Resolved at https://github.com/arcana-network/developer-dashboard/pull/119/commits/e6ab91a55b03a69772dd31b278c9f277636a5a1d

File: src/stores/toast.store.ts (hunk 5 above)
Comment (shrinathprabhu) [reply src/stores/toast.store.ts]: Resolved at https://github.com/arcana-network/developer-dashboard/pull/119/commits/382b9f1a46468ade06def818002345cb5100045c

File: src/stores/auth.store.ts (hunk 4 above)
Comment (shrinathprabhu) [reply src/stores/auth.store.ts]: Resolved at https://github.com/arcana-network/developer-dashboard/pull/119/commits/577670cd6194352399e16869992a428411a0320a

File: src/stores/url.store.ts (hunk 6 above)
Comment (shrinathprabhu) [reply src/stores/url.store.ts]: Resolved at https://github.com/arcana-network/developer-dashboard/pull/119/commits/a0b154b2272f25e889f7c83ab9a237d51398b217

File: src/stores/app.store.ts (hunk 3 above)
Comment (shrinathprabhu) [reply src/stores/app.store.ts]: Resolved at https://github.com/arcana-network/developer-dashboard/pull/119/commits/4a31142c17d9c5f2f9f006a070b7df8416eb0686

**Review by karthik-durai — APPROVED** [review]

**Pull request merged**